File: backend/dotnet-core/Project/Models/ResidenceReceipt.cs
using System.Text.Json.Serialization;

namespace Project.Models
{
    public class ResidenceReceipt
    {
        // Primary key
        public Guid ResidenceReceiptId { get; set; }   
        
        // Foreign keys
        public Guid PersonId { get; set; }

        // Properties
        public DateTime DateCreated { get; set; }
        public int Amount { get; set; }
        public string? Description { get; set; }

        // Navigation properties
        [JsonIgnore]
        public virtual Person? Person { get; set; }
        public virtual ICollection<ResidencePayment>? ResidencePayments { get; set; } = new List<ResidencePayment>();
    }
}

File: frontend/src/Models/ResidenceReceipt.js
export class ResidenceReceipt {
  constructor(
    dateCreated,
    amount,
    description,
    residencePayments,
    personId, 
    address,
    residenceReceiptId
  ) {
    this.description = description;
    this.dateCreated = dateCreated;
    this.amount = amount;
    this.residencePayments = residencePayments;
    this.personId = personId;
    this.address = address;
    this.residenceReceiptId = residenceReceiptId;
  }
}
